docs/examples/cancellation.nim

```diff
@@ -0,0 +1,42 @@
+import chronos/apps/http/httpclient
+
+proc cancellationExample() {.async.} =
+  # Simple cancellation
+  let future = sleepAsync(10.minutes)
+  future.cancelSoon()
+  # `cancelSoon` will not wait for cancellation to finish, so the `Future` could
+  # still be pending at this point.
+
+  # Wait for cancellation
+  let future2 = sleepAsync(10.minutes)
+  await future2.cancelAndWait()
+  # Using `cancelAndWait`, we know that future2 isn't pending anymore. However,
+  # it could have completed before cancellation happened (in which case, it
```

Comment:
As someone with very little understanding of async:
And what implications does it have for the future to hold a value if I want to cancel it?

Reply:
split into two examples to highlight this better